public/vislib/geoFilter.js

```diff
@@ -4,14 +4,66 @@ define(function (require) {
   const LAT_INDEX = 1;
   const LON_INDEX = 0;
 
-  return function GeoFilterFactory(Private) {
+  return function GeoFilterFactory(Private, confirmModal) {
     const _ = require('lodash');
     const queryFilter = Private(FilterBarQueryFilterProvider);
 
     function filterAlias(field, numBoxes) {
       return field + ": " + numBoxes + " geo filters"
     }
 
+    function _combineFilters(newFilter, existingFilter, field) {
+      let geoFilters = _.flatten([newFilter]);
+      let type = '';
+      if (_.has(existingFilter, 'bool.should')) {
+        geoFilters = geoFilters.concat(existingFilter.bool.should);
+        type = 'bool';
+      } else if (_.has(existingFilter, 'geo_bounding_box')) {
+        geoFilters.push({ geo_bounding_box: existingFilter.geo_bounding_box });
+        type = 'geo_bounding_box';
+      } else if (_.has(existingFilter, 'geo_polygon')) {
+        geoFilters.push({ geo_polygon: existingFilter.geo_polygon });
+        type = 'geo_polygon';
+      } else if (_.has(existingFilter, 'geo_shape')) {
+        geoFilters.push({ geo_shape: existingFilter.geo_shape });
+        type = 'geo_shape';
+      }
+
+      // Update method removed - so just remove old filter and add updated filter
+      const updatedFilter = {
+        bool: {
+          should: geoFilters
+        },
+        meta: existingFilter.meta
+      };
+      updatedFilter.meta.alias = filterAlias(field, geoFilters.length);
+      queryFilter.removeFilter(existingFilter);
+      queryFilter.addFilters([updatedFilter]);
+    }
+
+    function _overwriteFilters(newFilter, existingFilter, field, indexPatternName) {
```

Comment:
Looks like this function is a duplication of lines 89-104. Would mind removing lines 89-104 and replacing them with a call to this function?